### PR TITLE
Iv 310/watch details screen

### PR DESCRIPTION
**Pull request by msal96**

### CODE REVIEW

File: src/pods/catalog/web/apps/WatchCatalog/components/Gallery.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { Box, Image, Flex, Space, themeGet, Touchable } from '@ivoryio/kogaio'
+
+const images = [{ key: 'img1', src: 'https://images-na.ssl-images-amazon.com/images/I/71gdBQP%2BqGL._UL1500_.jpg'}, { key: 'img2', src: 'https://images-na.ssl-images-amazon.com/images/I/71gdBQP%2BqGL._UL1500_.jpg'}, { key: 'img3', src: 'https://images-na.ssl-images-amazon.com/images/I/71gdBQP%2BqGL._UL1500_.jpg'}, { key: 'img4', src: 'https://images-na.ssl-images-amazon.com/images/I/71gdBQP%2BqGL._UL1500_.jpg'}]
```
Comment (mihai-ro) [inline src/pods/catalog/web/apps/WatchCatalog/components/Gallery.js]: delete hardcoded images

File: src/pods/catalog/web/apps/WatchCatalog/components/Gallery.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            borderRadius={1}
+          />
+        </Flex>
+      </Space>
+      <Space px={{ xs: 2 }} py={{ xs: 2 }}>
+        <Flex flexWrap='wrap' width={{ xs: 1, md: 1 / 2, lg: 1 }}>
+          {
+            (imgList || images).map((imgSrc, index) => (
+              <Space
+                key={`detail-image-${index}`}
```
Comment (mihai-ro) [inline src/pods/catalog/web/apps/WatchCatalog/components/Gallery.js]: avoid using indexes as keys. your image probably contains an id.

File: src/pods/catalog/web/apps/WatchCatalog/components/Gallery.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                    <ImageContainer
+                      bg='white'
+                      alignItems='center'
+                      justifyContent='center'
+                      width={1}
+                    >
+                      <Image
+                        src={imgSrc}
+                        width={1}
+                        height={{ xs: 70, md: 156, lg: 133 }}
```
Comment (mihai-ro) [inline src/pods/catalog/web/apps/WatchCatalog/components/Gallery.js]: why the hardcoded values?

File: src/pods/catalog/web/apps/WatchCatalog/components/InfoTable.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  grid-template-columns: ${props => `${props.columnsWidth[0]}fr ${props.columnsWidth[1]}fr`};
+  grid-gap: 0;
+  & :last-child {
+    border-block-end: ${themeGet('borders.1')} ${themeGet('colors.pastel-blue')};
+  }
+  & :nth-last-child(2) {
+    border-block-end: ${themeGet('borders.1')} ${themeGet('colors.pastel-blue')};
+  }
+`
+const OptionName = styled(Typography)`
```
Comment (mihai-ro) [inline src/pods/catalog/web/apps/WatchCatalog/components/InfoTable.js]: Please add explicitly Typography-relevant styles; you should not be adding background and borders to Typography, but rather to a Typography's container

File: src/pods/catalog/web/apps/WatchCatalog/components/ProductSpecificationsMobile.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  Space,
+  Touchable
+} from '@ivoryio/kogaio'
+
+import { NavigationContext } from '../WatchCatalogEntry'
+import { TablesOnMobile } from '.'
+
+const ProductSpecificationsMobile = ({ ...props }) => {
+  const [activeSection, setActiveSection] = useState('details')
+  const { watchDetails: { data: { listingNumber, referenceNumber, brand, model, movement, year, gender, caliber: Caliber, case: Case, strap: Strap, description } } } = useContext(NavigationContext)
```
Comment (mihai-ro) [inline src/pods/catalog/web/apps/WatchCatalog/components/ProductSpecificationsMobile.js]: Please use camelCase naming for objects.

File: src/pods/catalog/web/apps/WatchCatalog/components/ProductSpecificationsMobile.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+            <Typography color='gunmetal' fontSize={3} fontWeight={0}>
+              Details
+            </Typography>
+            {
+              activeSection.includes('details') ? <Flex bg='brand' borderRadius={1} height='4px' /> : null
+            }
+          </Flex>
+        </Touchable>
+        <Space ml={{ xs: 8, md: 15 }}>
+          <Touchable onClick={() => setActiveSection('info&stats')}>
```
Comment (mihai-ro) [inline src/pods/catalog/web/apps/WatchCatalog/components/ProductSpecificationsMobile.js]: I suggest using a more common id format, with no special non-alphanumeric characters like ampersand. I suggest renaming it to info-stats

File: src/pods/catalog/web/apps/WatchCatalog/components/ProductSpecificationsMobile.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+            <Flex flexDirection='column'>
+              <Typography color='gunmetal' fontSize={3} fontWeight={0}>
+                Info & Stats
+              </Typography>
+              {
+                activeSection.includes('info&stats') ? <Flex bg='brand' borderRadius={1} height='4px' /> : null
```
Comment (mihai-ro) [inline src/pods/catalog/web/apps/WatchCatalog/components/ProductSpecificationsMobile.js]: same as above

Comment (mihai-ro) [reply src/pods/catalog/web/apps/WatchCatalog/components/ProductSpecificationsMobile.js]: Also, please fix this; we don't need an extra component for underlining the tab button.

I suggest you replace tab buttons with wrapped-in-Touchable Typography, that has a a border-bottom

File: src/pods/catalog/web/apps/WatchCatalog/components/ProductSpecificationsMobile.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+              }
+            </Flex>
+          </Touchable>
+        </Space>
+      </Flex>
+      <Space mt={{ xs: 4, md: 5 }}>
+        {activeSection.includes('details')
+          ? <Typography lineHeight='26px' color='gunmetal' fontSize={2} fontWeight={0}>{description}</Typography>
```
Comment (mihai-ro) [inline src/pods/catalog/web/apps/WatchCatalog/components/ProductSpecificationsMobile.js]: Careful with such random values as `lineHeight='26px;`I suggest you review this style and fix this; alternatively, use a typography variant.

File: src/pods/catalog/web/apps/WatchCatalog/components/ProductSpecificationsWeb.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
+import React, { Fragment, useContext } from 'react'
+import { Flex, Typography, Space } from '@ivoryio/kogaio'
+
+import { NavigationContext } from '../WatchCatalogEntry'
+import { InfoTable } from '.'
+
+const ProductSpecificationsWeb =  props => {
+  const { watchDetails: { data: { listingNumber, referenceNumber, brand, model, movement, year, gender, caliber: Caliber, case: Case, strap: Strap, description } } } = useContext(NavigationContext)
```
Comment (mihai-ro) [inline src/pods/catalog/web/apps/WatchCatalog/components/ProductSpecificationsWeb.js]: Same as Mobile version. use camelCase naming for objects.

File: src/pods/catalog/web/apps/WatchCatalog/components/ProductSpecificationsWeb.js [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+    Strap
+  }
+
+  const tableKeys = Object.keys(tableData)
+  
+  return (
+    <Flex flexDirection='column' {...props}>
+      <Typography id='details' color='gunmetal' fontSize={3} fontWeight={0}>Details</Typography>
+      <Space mt={2}>
+        <Typography lineHeight='26px' color='gunmetal' fontSize={1} fontWeight={0}>{description}</Typography>
```
Comment (mihai-ro) [inline src/pods/catalog/web/apps/WatchCatalog/components/ProductSpecificationsWeb.js]: Same as above; review lineHeight style.

File: src/pods/catalog/web/apps/WatchCatalog/screens/WatchDetails.js [hunk 11]
```diff
@@ -1,24 +1,81 @@
...
-          onClick={() => NavigateTo('watch-list')}
-        />
+      <Flex width={{ xs: 1, lg: 2 / 3}}>
+        <Space ml={{ xs: 1, md: 2, lg: 4 }} mt={{ xs: 2, md: 3, lg: 4 }}>
+          <BackButton onClick={goBack} />
+        </Space>
+      </Flex>
+      <Space mt={{ xs: 4, lg: 5 }}>
+        <Flex flexWrap='wrap' width={{ xs: 1, lg: 2 / 3 }}>
+          <Flex flexWrap='wrap' width={{ xs: 1, lg: 6 / 10 }}>
```
Comment (mihai-ro) [inline src/pods/catalog/web/apps/WatchCatalog/screens/WatchDetails.js]: You could simplify fractions

File: src/pods/catalog/web/apps/WatchCatalog/components/AddToCart.js [hunk 12]
```diff
@@ -0,0 +1,173 @@
...
+        {...props}
+      >
+        <Space pt={4} pb={{ xs: 4, lg: 0 }} top={0}>
+          <Card
+            zIndex={2}
+            width={1}
+            position={{ xs: 'block', lg: 'sticky' }}
+            bg='white'
+            boxShadow='card-simple'
+            display='flex'
```
[the remark this answers is not in the document]
Comment (msal96) [reply src/pods/catalog/web/apps/WatchCatalog/components/AddToCart.js]: It's not if I use mx='auto' for content-wrapper of the card

File: src/pods/catalog/web/apps/WatchCatalog/components/Gallery.js (hunk 2 above)
Comment (msal96) [reply src/pods/catalog/web/apps/WatchCatalog/components/Gallery.js]: Well, it's not

File: src/pods/catalog/web/apps/NewestWatches/NewestWatches.js [hunk 13]
```diff
@@ -110,23 +106,23 @@ const NewestWatches = () => {
...
-                      imgHeight='140px'
-                      onClick={() => {}}
-                    />
-                  </Box>
-                </Space>
-              ))
-            )}
+                items.map(({ id, brand, model, description, imgSrc }) => (
+                  <Space key={id} px={{ xs: 2, lg: 3 }}>
+                    <Box>
```
Comment (mihai-ro) [inline src/pods/catalog/web/apps/NewestWatches/NewestWatches.js]: Is the Box really necessary here?

File: src/pods/catalog/web/apps/WatchCatalog/WatchCatalogEntry.js [hunk 14]
```diff
@@ -1,19 +1,41 @@
...
+          setWatchDetails({ data, isFetching: false, error: null })
+        } else {
+          setWatchDetails(prevWatchDetails => ({ ...prevWatchDetails, isFetching: false, error: response.error }))
+        }
+      } catch (err) {
+        setWatchDetails(prevWatchDetails => ({ ...prevWatchDetails, isFetching: false, error: err }))
+      }
+    }
+
+    activeWatchId ? _fetchWatchDetails(activeWatchId) : setWatchDetails(prevWatchDetails => ({ ...prevWatchDetails, isFetching: true }))
```
Comment (mihai-ro) [inline src/pods/catalog/web/apps/WatchCatalog/WatchCatalogEntry.js]: This is some dangerous logic you are adding into a side effect in your Context Provider. you might want to rethink this logic and probably move it to somewhere else

File: src/pods/catalog/web/apps/WatchCatalog/components/AddToCart.js [hunk 15]
```diff
@@ -0,0 +1,164 @@
+import React, { useContext, useState } from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { Box, Button, Card, Flex, Hide, Icon, Space, themeGet, Touchable, Typography } from '@ivoryio/kogaio'
+
+import { NavigationContext } from '../WatchCatalogEntry'
+import { formatPrice } from '../services/helpers'
+
+const AddToCart = ({ ...props }) => {
```
Comment (mihai-ro) [inline src/pods/catalog/web/apps/WatchCatalog/components/AddToCart.js]: Looks like a pointless destructuring

File: src/pods/catalog/web/apps/WatchCatalog/components/AddToCart.js [hunk 16]
```diff
@@ -0,0 +1,164 @@
...
+  const [quantity, setQuantity] = useState(1)
+
+  const { watchDetails: {
+    data: {
+      brand,
+      model,
+      price
+    } } } = useContext(NavigationContext)
+
+  const _decrementQuantity = () => {
```
Comment (mihai-ro) [inline src/pods/catalog/web/apps/WatchCatalog/components/AddToCart.js]: As previously suggested, you could convert this into a one-liner.

File: src/pods/catalog/web/apps/WatchCatalog/components/InfoTable.js [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+import {
+  Box,
+  Typography,
+  themeGet,
+  Space
+} from '@ivoryio/kogaio'
+import { transformCamelToSentence } from '../services/helpers'
+
+const InfoTable = ({ columnsWidth, options, ...props }) => {
+  if (!options) {
```
Comment (mihai-ro) [inline src/pods/catalog/web/apps/WatchCatalog/components/InfoTable.js]: Move this just above the functional component's `return`

File: src/pods/catalog/web/apps/WatchCatalog/components/ProductSpecificationsMobile.js [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+    braceletMaterial: strap.braceletMaterial,
+    year,
+    gender
+  }
+  const tableData = { info: infoSectionData, caliber, watchCase, strap }
+  const tableKeys = Object.keys(tableData)
+
+  return (
+    <Flex flexDirection='column' {...props}>
+      <Flex>
```
Comment (mihai-ro) [inline src/pods/catalog/web/apps/WatchCatalog/components/ProductSpecificationsMobile.js]: Is this wrapper really needed? You might consider re-shaping <Tab /> for this manner.

File: src/pods/catalog/web/apps/WatchCatalog/components/SearchBox.js [hunk 19]
```diff
@@ -32,7 +43,7 @@ const SearchBox = ({ initialValue, searchWatches, ...rest }) => {
           <Touchable
```
Comment (mihai-ro) [inline src/pods/catalog/web/apps/WatchCatalog/components/SearchBox.js]: You might consider using <IconButton />

File: src/pods/catalog/web/apps/NewestWatches/NewestWatches.js (hunk 13 above)
Comment (msal96) [reply src/pods/catalog/web/apps/NewestWatches/NewestWatches.js]: Yes, it is

File: src/pods/catalog/web/apps/WatchCatalog/WatchCatalogEntry.js (hunk 14 above)
Comment (msal96) [reply src/pods/catalog/web/apps/WatchCatalog/WatchCatalogEntry.js]: You already updated this logic

File: src/pods/catalog/web/apps/WatchCatalog/components/AddToCart.js (hunk 15 above)
Comment (msal96) [reply src/pods/catalog/web/apps/WatchCatalog/components/AddToCart.js]: Not anymore :)) { isAwaitingData, ...props }

File: src/pods/catalog/web/apps/WatchCatalog/components/AddToCart.js (hunk 16 above)
Comment (msal96) [reply src/pods/catalog/web/apps/WatchCatalog/components/AddToCart.js]: done

File: src/pods/catalog/web/apps/WatchCatalog/components/ProductSpecificationsMobile.js (hunk 18 above)
Comment (msal96) [reply src/pods/catalog/web/apps/WatchCatalog/components/ProductSpecificationsMobile.js]: Yes, it's a wrapper for the tabs which has flexDirection='row' by default

File: src/pods/catalog/web/apps/WatchCatalog/components/SearchBox.js (hunk 19 above)
Comment (msal96) [reply src/pods/catalog/web/apps/WatchCatalog/components/SearchBox.js]: I replaced it with SquaredBox component

**Pull request merged**